packages/app/src/App.tsx

```diff
@@ -305,7 +305,7 @@ const routes = (
   </FlatRoutes>
 );
 
-export default app.createRoot(
+export default app.createRoot().render(
```

Comment:
This isn't a level of indirection that I think we want to add. The purpose of the current design of `.createRoot(el)` is to be able to traverse the app once and to know that it won't change over time. Eventually we'll want to get rid of the `.getProvider()` method too. The fact that this methods shares its name with the React method doesn't mean it needs to function the same way 😅. It's a method after all, closely tied to its instance.

This is also a breaking change that we won't want to do.

Honestly at this point I don't think we'll do anything with the `.getProvider()` method until we know the way forward for #18372, so for now I would recommend using that instead if you need to make runtime changes to the app structure. I could see something like a `.createMutableRoot()` being added down the line, but right now I think it's best to avoid changes that clutter the API surface.

Reply:
@Rugvip that sounds good; however, it might be a good idea to undeprecate `.getProvider()` since it is still required for testing purposes.

Reply:
I'd like to avoid un-deprecating it since almost all use-cases don't actually need it. Anyone that needs to use the old `.getProvider()` method to dynamically update the app can safely keep using it for now. I'd like to know a bit more about how #18372 continues forward before making changes to these core APIs to avoid noise and potential overhead of refactoring the app core.